### ISSUE TITLE
Suggestion: Update regex pattern for validating file paths

### ISSUE THREAD

**Issue by Skeb1ns** [MSEndpointMgr/IntuneWin32App#61]
Maybe it's a good idea to update the regex pattern for validating filepaths which for example is used in New-IntuneWin32AppIcon.

The current pattern used is "^[A-Za-z]{1}:\\\w+\\\w+" which doesn't account for dashes in paths. Was pulling my hair out for the last 30 minutes figuring out why the cmdlet didn't accept the path because it is perfectly valid ;).

**Comment by Arcath** [comment]
Just been hitting the same thing.

It's worse than not accepting punctuation, you also have to use a folder on the disk not the root, and you can't have spaces in the folder path anywhere.

**Comment by NickolajA** [comment]
This makes total sense, I'll look into fixing that. If any of you have a better regex, I'm all ears.

**Comment by NickolajA** [comment]
> The same for me - I had the same issues, when tried to run these commandlets from Linux machine and one of big issues is regex rules for checking paths - because on WSL I don't have paths like "C:....". So my suggestion to change somehow these validating rules.

Out of curiosity, why are you not running this module in a PowerShell console directly in Windows, but in WSL? Although it's not stated anywhere for the module, it's not a scenario that I'm testing at all when developing this module.

**Comment by NickolajA** [comment]
I'm going to with the following regex, if nobody has a better one or objections:

```
$Pattern = '^[a-z]:\\(?:[^\\/:*?"<>|\r\n]+\\)*[^\\/:*?"<>|\r\n]*$'
"C:\Folder" -match $Pattern
"C:\Folder\SubFolder" -match $Pattern
"C:\Folder\SubFolder\File.intunewin" -match $Pattern
"C:\Folder\SubFolder\File.png" -match $Pattern
"C:\Folder Space" -match $Pattern
"C:\Folder Space\SubFolder" -match $Pattern
"C:\Folder Space\SubFolder\File.intunewin" -match $Pattern
"C:\Folder Space\SubFolder\File.png" -match $Pattern
"C:\Folder.Dot" -match $Pattern
"C:\Folder.Dot\SubFolder" -match $Pattern
"C:\Folder.Dot\SubFolder\File.intunewin" -match $Pattern
"C:\Folder.Dot\SubFolder\File.png" -match $Pattern
```

All of the above renders True.

**Comment by TheQuiz03** [comment]
Including UNC paths would be much appreciated. https://github.com/MSEndpointMgr/IntuneWin32App/issues/17

**Comment by NickolajA** [comment]
> Including UNC paths would be much appreciated. #17

While I'm not a fan of it, as it potentially brings new issues into the mix (since the code has never been tested with UNC paths), wouldn't it be easier for you to simply mount your network share instead?

**Comment by g3rhard** [comment]
> > The same for me - I had the same issues, when tried to run these commandlets from Linux machine and one of big issues is regex rules for checking paths - because on WSL I don't have paths like "C:....". So my suggestion to change somehow these validating rules.
> 
> Out of curiosity, why are you not running this module in a PowerShell console directly in Windows, but in WSL? Although it's not stated anywhere for the module, it's not a scenario that I'm testing at all when developing this module.

Just because I want to put it into pipelines (Jenkins) and as for me better will be to run these scripts from Linux.
I can say, that almost all comandlets from this repo is cross platform (good job, really, I can run it on Powershell 7 in Linux), except part with regex. On my system I just commented sections with regex, as workaround, but I can't say that is good way to solve this.

**Comment by TheQuiz03** [comment]
Could it be moved from a rexex to using test-path?

**Comment by NickolajA** [comment]
> 

That makes total sense, although I've used it in Pipelines on a Windows host as I don't really know Linux that much. I'm leaning towards simply just scrapping the regex validation, and only keep the Test-Path piece of the code. Would you be able to validate a pre-release of 1.4.0 that it works as expected in a cross-platform scenario, if I made it available?

**Comment by g3rhard** [comment]
Yes, I'll try to check it as soon as possible and will provide results in this issue, thanks for your help 😄 👍

**Comment by NickolajA** [comment]
1.4.0 (RC) source code is now in the repo. Regex validation has been replaced with Test-Path detection instead. Please test it out to see if it works :)

**Comment by g3rhard** [comment]
Hello, @NickolajA!
1.4.0 works like a charm (tested on Windows Powershell and Linux Powershell) - I also fixed other issues, with requirements, many thanks for your support :)

```
Name                           Value
----                           -----
PSVersion                      7.3.2
PSEdition                      Core
GitCommitId                    7.3.2
OS                             Linux 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by NickolajA** [comment]
That's awesome, thanks for validating!

**Comment by NickolajA** [comment]
I'll mark this as fixed now, as this will be added to 1.4.0 that should come out in the next few days.
